### PR TITLE
Upgrade: Add .get() and .count() methods

### PR DESCRIPTION
**Pull request by dassantoss**
#### Description
This pull request aims to enhance the `DBStorage` and `FileStorage` classes by introducing two new methods: `get()` and `count()`. These changes have been made within the `storage_get_count` branch.

The `get()` method retrieves an object based on its class and ID, returning the object or `None` if not found. On the other hand, the `count()` method counts the number of objects in storage, either matching a given class or all objects if no class is specified.

In addition to implementing the methods, new test cases have been added to validate the functionality of these changes on each storage engine.

#### Changes Made
- Added `get()` method to retrieve objects from storage based on class and ID.
- Added `count()` method to count the number of objects in storage, optionally filtered by class.
- Added new test cases for the `get()` and `count()` methods on both storage engines.

#### Testing
Test cases have been provided to validate the correct functionality of the new methods on both the `DBStorage` and `FileStorage` engines.

#### Review and Merging
I kindly request a code review for this pull request from my fellow developers. Once the changes are reviewed and approved, I would appreciate it if someone could merge the pull request.

#### Additional Notes
Please refrain from deleting the `storage_get_count` branch, as a manual review for grading purposes will be conducted using this branch.

Thank you for your attention.

### CODE REVIEW

File: models/engine/db_storage.py [hunk 1]
```diff
@@ -74,3 +74,14 @@ def reload(self):
     def close(self):
         """call remove() method on the private session attribute"""
         self.__session.remove()
+
+    def get(self, cls, id):
+        """retrieve one object"""
```
Comment (TaniaHillock) [inline models/engine/db_storage.py:79]: That is true, we really needed this method.

File: tests/test_models/test_engine/test_db_storage.py [hunk 2]
```diff
@@ -86,3 +86,11 @@ def test_new(self):
     @unittest.skipIf(models.storage_t != 'db', "not testing db storage")
     def test_save(self):
         """Test that save properly saves objects to file.json"""
+
+    @unittest.skipIf(models.storage_t != 'db', "not testing db storage")
+    def test_get(self):
+        """Test that get returns the correct object"""
```
Comment (TaniaHillock) [inline tests/test_models/test_engine/test_db_storage.py:92]: user = User(name="User1")
        user.save()
        self.assertEqual(models.storage.get("User", user.id), user)

**Review by TaniaHillock — CHANGES_REQUESTED** [review]
We need to ensure that the storage system correctly retrieves objects by their ID. By example:

File: tests/test_models/test_engine/test_db_storage.py (hunk 2 above)
Comment (dassantoss) [reply tests/test_models/test_engine/test_db_storage.py:92]: Sure I had modified the file and made these changes

**Comment by AppEPTI** [comment]
Good job.

**Pull request merged**